python/cudf/cudf/utils/utils.py

```diff
@@ -37,6 +39,65 @@
 }
 
 
+# The test root is set by pytest to support situations where tests are run from
+# a source tree on a built version of cudf.
+NO_EXTERNAL_ONLY_APIS = os.getenv("NO_EXTERNAL_ONLY_APIS")
+_CUDF_TEST_ROOT = os.getenv("_CUDF_TEST_ROOT")
+
+if NO_EXTERNAL_ONLY_APIS in ("True", "1", "TRUE"):
```

Comment:
I'd prefer to let this trigger any time the variable is _set_, rather than on some indeterminate set of values. If it is unset, `os.getenv("NONEXISTENT_VARIABLE")` returns `None`.
```suggestion
if NO_EXTERNAL_ONLY_APIS:
```